test/boost/mutation_query_test.cc

```diff
@@ -492,6 +493,104 @@ SEASTAR_TEST_CASE(test_result_row_count) {
+
+        tests::reader_concurrency_semaphore_wrapper semaphore;
+
+        auto slice = partition_slice_builder(*s).build();
+        auto rev_slice = partition_slice_builder(*s).reversed().build();
+        auto src = make_source(muts);
+
+        utils::chunked_vector<partition> parts;
+        parts.push_back(partition(3, freeze(muts[0])));
+        parts.push_back(partition(3, freeze(muts[1])));
+        parts.push_back(partition(3, freeze(muts[2])));
+        reconcilable_result rr(9, std::move(parts), query::short_read::yes);
+
+        {
+            query::result r = to_data_query_result(rr, s, slice, 1, inf32);
+            testlog.trace("r = {}", r.pretty_print(s, slice));
+            BOOST_REQUIRE_EQUAL(r.row_count().value(), 1);
+            BOOST_REQUIRE(r.short_read_pos());
+            auto [pkey, ckey] = *r.short_read_pos();
+            BOOST_REQUIRE(pkey.equal(*s, pkeys[0].key()));
+            BOOST_REQUIRE(ckey);
+            BOOST_REQUIRE(ckey->equal(*s, ss.make_ckey(1)));
+        }
+
+        {
+            query::result r = to_data_query_result(rr, s, slice, inf32, inf32);
+            testlog.trace("r = {}", r.pretty_print(s, slice));
+            BOOST_REQUIRE_EQUAL(r.row_count().value(), 9);
+            BOOST_REQUIRE(r.short_read_pos());
+            auto [pkey, ckey] = *r.short_read_pos();
+            BOOST_REQUIRE(pkey.equal(*s, pkeys[2].key()));
+            BOOST_REQUIRE(ckey);
+            BOOST_REQUIRE(ckey->equal(*s, ss.make_ckey(9)));
+        }
+
+        {
+            query::result r = to_data_query_result(rr, s, slice, inf32, 2);
+            testlog.trace("r = {}", r.pretty_print(s, slice));
+            BOOST_REQUIRE_EQUAL(r.row_count().value(), 6);
+            BOOST_REQUIRE(r.short_read_pos());
+            auto [pkey, ckey] = *r.short_read_pos();
+            BOOST_REQUIRE(pkey.equal(*s, pkeys[1].key()));
+            BOOST_REQUIRE(ckey);
+            BOOST_REQUIRE(ckey->equal(*s, ss.make_ckey(6)));
+        }
+
+        {
+            query::result r = to_data_query_result(rr, s, rev_slice, 1, inf32);
+            testlog.trace("r = {}", r.pretty_print(s, rev_slice));
+            BOOST_REQUIRE_EQUAL(r.row_count().value(), 1);
+            BOOST_REQUIRE(r.short_read_pos());
+            auto [pkey, ckey] = *r.short_read_pos();
+            BOOST_REQUIRE(pkey.equal(*s, pkeys[0].key()));
+            BOOST_REQUIRE(ckey);
+            BOOST_REQUIRE(ckey->equal(*s, ss.make_ckey(3)));
+        }
+
+        {
+            query::result r = to_data_query_result(rr, s, rev_slice, inf32, inf32);
+            testlog.trace("r = {}", r.pretty_print(s, rev_slice));
+            BOOST_REQUIRE_EQUAL(r.row_count().value(), 9);
+            BOOST_REQUIRE(r.short_read_pos());
+            auto [pkey, ckey] = *r.short_read_pos();
+            BOOST_REQUIRE(pkey.equal(*s, pkeys[2].key()));
+            BOOST_REQUIRE(ckey);
+            BOOST_REQUIRE(ckey->equal(*s, ss.make_ckey(7)));
+        }
+    });
+}
```

Comment:
Does this test case use any tombstones? It would be most interesting to verify what it returns when the page ends in the middle of a long range of tombstones.

Reply:
Right, especially because as I commented above, it seems your wrapper doesn't save the position when seeing a tombstone, so I don't understand if it does what you expect or not.

Reply:
partition tombstone doesn't affect page boundary position. range tombstones are not supported as a boundary too. Row tombstones are just rows in our model, so it's enough to test with rows.